cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/ProgramRunDispatcherTask.java

```diff
@@ -68,13 +68,18 @@ public void run(RunnableTaskContext context) throws Exception {
     ProgramRunDispatcherContext dispatcherContext = new ProgramRunDispatcherContext(
       GSON.fromJson(context.getParam(), ProgramRunDispatcherContext.class));
     context.setCleanupTask(dispatcherContext::executeCleanupTasks);
+    ProgramRunId programRunId =
+      dispatcherContext.getProgramDescriptor().getProgramId().run(dispatcherContext.getRunId());
 
     ProgramRunDispatcher dispatcher = injector.getInstance(InMemoryProgramRunDispatcher.class);
-    ProgramController programController = dispatcher.dispatchProgram(dispatcherContext);
+    ProgramController programController;
+    try {
+      programController = dispatcher.dispatchProgram(dispatcherContext);
+    } catch (Exception e) {
+      throw new Exception("Failure while dispatching program run " + programRunId, e);
```

Comment:
nit: change to `Failed to dispatch program run`

Reply:
Done